### PR TITLE
DTFS2-5228 - Split helper, macros for about-facility & refactor validation

### PR DESCRIPTION
**Pull request by Zainzzkk**
# Changes made:

* Split helper function into 3 helper files: deal-helpers, facility-helpers and general helpers
* Separated mock helper files into applications, facilities and response
* Passed body, params and query to validation for about-facility-change as compared to passing res and req
* Updated relevant tests
* Added unissued facility changed to clone deal
* Added macros for about-facility

### CODE REVIEW

File: gef-ui/server/utils/MOCKS/MOCK_FACILITIES.js [hunk 1]
```diff
@@ -0,0 +1,178 @@
+const CONSTANTS = require('../../constants');
```
Comment (ttbarnes) [inline gef-ui/server/utils/MOCKS/MOCK_FACILITIES.js:1]: Personal preference, totally up to you, but I think lowercase filenames is more friendly 😍 What do you think?

**Review by ttbarnes — APPROVED** [review]
Beautiful

File: gef-ui/server/utils/MOCKS/MOCK_FACILITIES.js (hunk 1 above)
Comment (Zainzzkk) [reply gef-ui/server/utils/MOCKS/MOCK_FACILITIES.js:1]: changed

File: gef-ui/server/controllers/unissued-facilities/index.test.js [hunk 2]
```diff
@@ -205,24 +204,41 @@ describe('postChangeUnissuedFacility()', () => {
...
 
-    mockRequest.body['cover-start-date-day'] = format(threeMonthsAndOneDayFromNow, 'd');
-    mockRequest.body['cover-start-date-month'] = format(threeMonthsAndOneDayFromNow, 'M');
-    mockRequest.body['cover-start-date-year'] = format(threeMonthsAndOneDayFromNow, 'yyyy');
+    await postChangeUnissuedFacility(mockRequest, mockResponse);
 
-    mockRequest.body['cover-end-date-day'] = format(oneYearFromNow, 'd');
-    mockRequest.body['cover-end-date-month'] = format(oneYearFromNow, 'M');
-    mockRequest.body['cover-end-date-year'] = format(oneYearFromNow, 'yyyy');
+    // console.log(mockResponse.render);
```
Comment (ttbarnes) [inline gef-ui/server/controllers/unissued-facilities/index.test.js:213]: remove but can be done after this PR is merged as it's blocking others :D

**Review by ttbarnes — APPROVED** [review]
LGTM, nice one!

File: gef-ui/server/controllers/unissued-facilities/index.test.js (hunk 2 above)
Comment (Zainzzkk) [reply gef-ui/server/controllers/unissued-facilities/index.test.js]: removed.  Had linting error anyways so removed as part of fix

**Review by ttbarnes — DISMISSED** [review]

**Review by abhi-markan — DISMISSED** [review]

**Review by abhi-markan — APPROVED** [review]

**Review by ttbarnes — APPROVED** [review]

**Pull request merged**